base/reducedim.jl

```diff
@@ -112,7 +112,7 @@ function reducedim_init(f, op::typeof(*), A::AbstractArray, region)
 end
 function _reducedim_init(f, op, fv, fop, A, region)
     T = promote_union(eltype(A))
-    if applicable(zero, T)
+    if T !== Any && applicable(zero, T)
```

Comment:
This one isn't totally satisfying, but I had to add it due to the `zero(::Type{Any}) = throw(MethodError(zero, (Any,)))` method, which is in turn needed to replace a `StackOverflowError` with a more explicit error. I wonder whether `applicable` should treat methods that throw unconditionally as if they didn't exist.

Reply:
> I wonder whether applicable should treat methods that throw unconditionally as if they didn't exist.

That would be good, although probably just the ones which throw a `MethodError` (or some other special error, e.g. `NotImplementedError`).